modules/http/http.c

```diff
@@ -491,6 +223,324 @@ http_dd_set_timeout(LogDriver *d, glong timeout)
+_format_persist_name(const LogPipe *s)
+{
+  const HTTPDestinationDriver *self = (const HTTPDestinationDriver *)s;
+  static gchar persist_name[1024];
+
+  if (s->persist_name)
+    g_snprintf(persist_name, sizeof(persist_name), "http.%s", s->persist_name);
+  else
+    g_snprintf(persist_name, sizeof(persist_name), "http(%s,)", self->url);
+
+  return persist_name;
+}
+
+static const gchar *
+_format_stats_instance(LogThreadedDestDriver *s)
+{
+  static gchar stats[1024];
+
+  HTTPDestinationDriver *self = (HTTPDestinationDriver *) s;
+
+  g_snprintf(stats, sizeof(stats), "http,%s", self->url);
+
+  return stats;
+}
+
+
+/* Set up options that are static over the course of a single configuration,
+ * request specific options will be set separately
+ */
+static void
+_setup_static_options_in_curl(HTTPDestinationDriver *self)
+{
+  curl_easy_reset(self->curl);
+
+  curl_easy_setopt(self->curl, CURLOPT_WRITEFUNCTION, _curl_write_function);
+
+  curl_easy_setopt(self->curl, CURLOPT_URL, self->url);
+
+  if (self->user)
+    curl_easy_setopt(self->curl, CURLOPT_USERNAME, self->user);
+
+  if (self->password)
+    curl_easy_setopt(self->curl, CURLOPT_PASSWORD, self->password);
+
+  if (self->user_agent)
+    curl_easy_setopt(self->curl, CURLOPT_USERAGENT, self->user_agent);
+
+  if (self->ca_dir)
+    curl_easy_setopt(self->curl, CURLOPT_CAPATH, self->ca_dir);
+
+  if (self->ca_file)
+    curl_easy_setopt(self->curl, CURLOPT_CAINFO, self->ca_file);
+
+  if (self->cert_file)
+    curl_easy_setopt(self->curl, CURLOPT_SSLCERT, self->cert_file);
+
+  if (self->key_file)
+    curl_easy_setopt(self->curl, CURLOPT_SSLKEY, self->key_file);
+
+  if (self->ciphers)
+    curl_easy_setopt(self->curl, CURLOPT_SSL_CIPHER_LIST, self->ciphers);
+
+  curl_easy_setopt(self->curl, CURLOPT_SSLVERSION, self->ssl_version);
+  curl_easy_setopt(self->curl, CURLOPT_SSL_VERIFYHOST, self->peer_verify ? 2L : 0L);
+  curl_easy_setopt(self->curl, CURLOPT_SSL_VERIFYPEER, self->peer_verify ? 1L : 0L);
+
+  curl_easy_setopt(self->curl, CURLOPT_DEBUGFUNCTION, _curl_debug_function);
+  curl_easy_setopt(self->curl, CURLOPT_VERBOSE, (long) (debug_flag || trace_flag));
+
+  curl_easy_setopt(self->curl, CURLOPT_TIMEOUT, self->timeout);
+
+  if (self->method_type == METHOD_TYPE_PUT)
+    curl_easy_setopt(self->curl, CURLOPT_CUSTOMREQUEST, "PUT");
+}
+
+
+static struct curl_slist *
+_add_header(struct curl_slist *curl_headers, const gchar *header, const gchar *value)
+{
+  GString *buffer = scratch_buffers_alloc();
+
+  g_string_append(buffer, header);
+  g_string_append(buffer, ": ");
+  g_string_append(buffer, value);
+  return curl_slist_append(curl_headers, buffer->str);
+}
+
+static struct curl_slist *
+_format_request_headers(HTTPDestinationDriver *self, LogMessage *msg)
+{
+  struct curl_slist *headers = NULL;
+  GList *l;
+
+  headers = _add_header(headers, "Expect", "");
+  if (msg)
+    {
+      /* NOTE: I have my doubts that these headers make sense at all.  None of
```

Comment:
Optional request:
Good topic, thanks for opening it! Maybe these can be removed, especially if we can provide alternative? For example if headers could handle templates? For now they don't.
Still I would **not** commit this comment into the code, but move the discussion into an issue or PR. Because discussion inside the code makes participating hard. As this would require participants to open a pr to add a comment.

Reply:
While I agree that work items would need to be scheduled/filed, I think notes like this help navigation and subsequent adjustment of functionality for someone coming later to read the code.

For example, at the end I could easily retain this functionality, but only in case batching is disabled. With that I dont see removing that code is priority. Should we need to change the http module in the future, this comment helps to add a +1 to cases where this functionality becomes a bottleneck for further improvements, bumping the motivation to actually remove it. Adding a low prio ticket that will never get scheduled will not achieve the same, as it is disconnected from the code. We could add the ticket and a reference from the code, which is a bit of an improvement, I tend to agree. I was lazy not filing, but I dont think the ticket would help that much.